agent/testagent.go

```diff
@@ -123,79 +131,109 @@ func (a *TestAgent) Start(t *testing.T) *TestAgent {
 		}
 		name = strings.Replace(name, "/", "_", -1)
 		d, err := ioutil.TempDir(TempDir, name)
```

Comment:
Sidenote: the fact that it's commented out _by default_ is a fun nightmare to reverse. Almost no tests need to test resuming one server from the state of another instance of the server (a valid test, but not common). Rather than make the cleanup behavior opt-out, this change made it opt-in which means nothing really cleans up at all, and `/tmp` is full of litter which is fun.

I guess I was concerned mostly in the case where within one call of `TestAgent.Start()` you create a temp directory and then early return at any point before the eventual cleanup that happens when `Agent.Start()` fails. In that case you'd leak the temp directory.

Reply:
ok, i think i've got this one covered now. code push coming shortly.